### PR TITLE
feat(httpoptions): allow configuring httpOptions.rejectUnAuthorized from environment variables

### LINKED ISSUE #99: Allow configuring the value for  httpOptions.rejectUnauthorized via Environment variables
When using unleash proxy docker image, it is not possible to configure it to allow self signed certificates, I have tried using the environment variables `NODE_EXTRA_CA_CERTS `and `NODE_TLS_REJECT_UNAUTHORIZED`, both doesn't appear to stop the "request to https://gitlab.example.com/api/v4/feature_flags/unleash/000000/client/register failed, reason: self signed certificate in certificate chain" error.

The only workaround is to run the unleash proxy as a node js application and adding `httpOptions: { rejectUnauthorized: false }` in the createApp configuration

### Example

```
const app = createApp({
    unleashUrl: 'https://gitlab.example.com/api/v4/feature_flags/unleash/000000',
    unleashInstanceId: '0xx-0x0_xxXXXXX0_xx0',
    unleashApiToken: 'xxxxx-xxxxxxxxxxxxxxxxxx',
    unleashAppName: "example",
    clientKeys: ['some-secret'],
    proxyPort: 3000,
    **httpOptions: { rejectUnauthorized: false }**
});
```

If the value for httpOptions.rejectUnauthorized can be read from the Environment variables, then we can use the unleash proxy with the provided docker image

### PR DESCRIPTION
**Pull request by vkbandi**
## About the changes
These changes allow setting the httpOptions.rejectUnAuthorized from environment variables

Closes #99

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -231,6 +231,20 @@ function loadCorsOptions(option: IProxyOption): CorsOptions {
...
 
+function loadHttpOptions(option: IProxyOption): HttpOptions {
+    if (option.httpOptions) {
+        return option.httpOptions;
+    }
+
+    if (process.env.HTTP_OPTIONS_REJECT_UNAUTHORIZED) {
+        return {
+            rejectUnauthorized: safeBoolean(process.env.HTTP_OPTIONS_REJECT_UNAUTHORIZED, false)
+        };
```
Comment (Tymek) [inline src/config.ts:242]: This can return `{ httpOptions: {...} }` or `{}` in order to simplify logic in L:344
(to be `...loadHttpOptions(option),`)

**Review by Tymek — CHANGES_REQUESTED** [review]
Thanks for your contribution! 👍🏻 
Can you also add this ENV option to [README #available-options section](/Unleash/unleash-proxy#available-options) for consistency?

**Comment by vkbandi** [comment]
> Thanks for your contribution! 👍🏻 Can you also add this ENV option to [README #available-options section](/Unleash/unleash-proxy#available-options) for consistency?

Made this change

File: src/config.ts (hunk 1 above)
Comment (vkbandi) [reply src/config.ts]: Done this change

**Comment by vkbandi** [comment]
@Tymek - can you please check this PR

**Review by ivarconr — APPROVED** [review]

**Pull request merged**